src/tests/dotnet-trace/ChildProcessTests.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.NETCore.Client;
+using System;
+using Xunit;
+using Xunit.Abstractions;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+
+    public class ChildProcessTests
+    {
+        // Pass ITestOutputHelper into the test class, which xunit provides per-test
+        public ChildProcessTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+        }
+
+        private ITestOutputHelper OutputHelper { get; }
+
+        private void LaunchDotNetTrace(string command, out int exitCode, out string stdOut, out string stdErr)
+        {
+            string dotnetTracePathWithArgs = CommonHelper.GetTraceePathWithArgs(traceeName: "dotnet-trace").Replace("net5.0", "netcoreapp2.1");
+            ProcessStartInfo startInfo = new ProcessStartInfo(CommonHelper.HostExe, $"{dotnetTracePathWithArgs} {command}");
+
+            OutputHelper.WriteLine($"Launching: {startInfo.FileName} {startInfo.Arguments}");
+            startInfo.RedirectStandardInput = true;
+            startInfo.RedirectStandardError = true;
+            startInfo.RedirectStandardOutput = true;
+
+            using (Process process = Process.Start(startInfo))
+            {
+                bool processExitedCleanly = process.WaitForExit(10000);
+                if (!processExitedCleanly)
+                {
+                    process.Kill();
```

Comment:
@davidwrighton I think this is where the exit code 137 is coming from in those CI test failures. It's possible that the CI machines are sufficiently slow that they don't finish within 10s and the process.Kill() sends SIGKILL which results in the exit code 137.

Reply:
No, I don't think that this particular error code is coming from as here would be a different output from the failures, but it is a good point that I should print to the test log if this Kill is ever used as it would not be good if it was failing here often.